src/transformations/transformType.ts

```diff
@@ -0,0 +1,32 @@
+import { TransformableResource } from '@/schemas/configuration-schema';
+
+export function narrowType(records: Record<string, never>[], resource: TransformableResource) {
+    return records.map(
+        (record) =>
+            Object.fromEntries(
+                Object.entries(record).map(([key, value]) => {
+                    const stringValue = String(value).toLowerCase();
+                    if (stringValue === 'true') {
+                        return [key, true];
+                    }
+                    if (stringValue === 'false') {
+                        return [key, false];
+                    }
+                    const parsedValue =
+                        resource.numberFormat === 'en'
+                            ? Number(stringValue)
+                            : parseGermanNumberToInternationalFormat(stringValue);
+                    if (!Number.isNaN(parsedValue)) {
+                        return [key, parsedValue];
+                    }
+                    return [key, value];
+                }),
+            ) as Record<string, string | number | boolean>,
+    );
+}
+
+function parseGermanNumberToInternationalFormat(value: string) {
```

Comment:
We can probably simplify this function:
`const parsedValue = Number(value.replace('.', '').replace(',', '.'));`
`return !Number.isNaN(parsedValue) ? parsedValue : Number.NaN;`

Reply:
Good point